### ISSUE TITLE
💅 `useExpect` produces false negatives for Vitest asymmetric test matchers/`expect.extend`

### ISSUE THREAD

**Issue by Bertie690** [biomejs/biome#9174]
### Environment information

N/A

### Rule name

`lint/nursery/useExpect`

### Playground link

https://biomejs.dev/playground/?lineWidth=120&indentStyle=space&lintRules=useExpect&analyzerFixMode=safeAndUnsafeFixes&code=aQB0ACgAIgArACIALAAgACgAKQAgAD0APgAgAHsACgAgACAAZQB4AHAAZQBjAHQALgBzAHQAcgBpAG4AZwBDAG8AbgB0AGEAaQBuAGkAbgBnACgAJwBmAG8AbwAnACkACgB9ACkACgAKAGkAdAAoACIAbgBvAHQAIABhAG4AIABhAHMAcwBlAHIAdABpAG8AbgAiACwAIAAoACkAIAA9AD4AIAB7AAoAIAAgAGUAeABwAGUAYwB0AC4AZQB4AHQAZQBuAGQAKAB7AG0AeQBGAG8AbwBNAGEAdABjAGgAZQByAH0AKQAKAH0AKQA%3D

### Expected result

This should produce an error - `expect.stringContaining` is not an assertion and should not be treated as one - it is a vitest asymmetric matcher whose only purpose is to be passed to an `expect` call or similar.
As for `expect.extend`, it augments Vitest's `extend` syntax with the provided function, and is similarly not actually testing anything.

NB: From what I can tell, there's no easy way to tell what asymmetric matchers a consumer may have defined with [`expect.extend`](https://vitest.dev/api/expect.html#expect-extend), as it will add both asymmetric matchers AND custom assertion functions with the same name.

```ts
it("a", () => {
  expect.extend({myFooMatcher});

  // these are both valid syntaxes
  expect(new Foo()).toBe(expect.myFooMatcher());
  expect(new Foo()).myFooMatcher();
});
```

If nothing else, it may be wise to mention this limitation inside the rule itself.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by omar-y-abdi** [comment]
I'm working on a fix for this. The root cause is in `is_expect_expression` in `playwright.rs` — the `JsStaticMemberExpression` branch unconditionally recurses on any `expect.XYZ` pattern without distinguishing assertion-qualifying members (`soft`, `poll`, `assertions`, `hasAssertions`) from asymmetric matchers and utilities.
